formal-spec/Leios/SimpleSpec.agda

```diff
@@ -115,45 +121,62 @@ data _⇀⟦_⟧_ : Maybe LeiosState → LeiosInput → LeiosState × LeiosOutpu
        ────────────────────────────────────
        just s ⇀⟦ FTCH-LDG ⟧ (s , FTCH-LDG l)
 
-  -- TODO: Base chain
+  -- Base chain
+  --
+  -- Note: Submitted data to the base chain is only taken into account
+  --       if the party submitting is the block producer on the base chain
+  --       for the given slot
+
+  Base₁ : ∀ {txs} → let open LeiosState s in
+        ──────────────────────────────────────────────────────────────────
+        just s ⇀⟦ SUBMIT (inj₂ txs) ⟧ (record s { MemPool = txs } , EMPTY)
+
+  Base₂a : ∀ {bs bs' eb} → let open LeiosState s in
+         ∙ eb ∈ filterˢ (λ eb → isVote2Certified s eb × eb ∈ᴮ slice L slot 2) (fromList EBs)
+         ∙ bs BF.⇀⟦ B.SUBMIT (inj₁ eb) ⟧ (bs' , B.EMPTY)
+         ────────────────────────────────────────────────────────────────────────────────────
+         just s ⇀⟦ SUBMIT (inj₁ eb) ⟧ (s , EMPTY)
+
+  Base₂b : ∀ {bs bs'} → let open LeiosState s renaming (MemPool to txs) in
+         ∙ ∅ˢ ≡ filterˢ (λ eb → isVote2Certified s eb × eb ∈ᴮ slice L slot 2) (fromList EBs)
```

Comment:
```suggestion
         ∙ ∅ ≡ filterˢ (λ eb → isVote2Certified s eb × eb ∈ᴮ slice L slot 2) (fromList EBs)
```
There's a `HasEmptySet` typeclass that provides this, so you don't need that `ˢ`. Also, using `≡` is fine here because we're comparing with the empty set, but in general you need to use `≡ᵉ` for the 'correct' set equality.